components/style/counter_style/mod.rs

```diff
@@ -482,7 +481,7 @@ fn parse_bound<'i, 't>(
     let location = input.current_source_location();
     match *input.next()? {
         Token::Number { int_value: Some(v), .. } => {
```

Comment:
This should stop matching on `Token`, and use `Integer::parse` instead.

Reply:
To use Integer::parse we need to have the context, which is not available in this scope. Also how will we change test for CounterBound::Infinite here?

Reply:
The caller of this function has it though, so we should be able to start passing it to this function.

Reply:
Regarding `infinite`, we should do something like:

```rust
if let Ok(integer) = input.try(|input| Integer::parse(context, input)) {
    return Ok(CounterBound::Integer(integer));
}
input.expect_ident_matching("infinite")?;
Ok(CounterBound::Infinite)
```

After this change we need to add a test for `calc()` being allowed in counter-style, with something like: `range: infinite calc(1)` or something like that.

Thanks!